Entities/Vehicles/Common/VehicleCommon.as

```diff
@@ -208,22 +218,38 @@ int server_LoadAmmo(CBlob@ this, CBlob@ ammo, int take, VehicleInfo@ v)
 
 void RecountAmmo(CBlob@ this, VehicleInfo@ v)
 {
-	int ammoStocked = v.loaded_ammo;
-	const string ammoName = v.ammo_name;
-	for (int i = 0; i < this.getInventory().getItemsCount(); i++)
+	for(int i = 0; i < v.ammo_types.size(); ++i)
 	{
-		CBlob@ invItem = this.getInventory().getItem(i);
-		if (invItem.getName() == ammoName)
+		AmmoInfo@ current_ammo = v.ammo_types[i];
+		int ammoStocked = current_ammo.loaded_ammo;
+		const string ammoName = current_ammo.ammo_name;
+		for (int i = 0; i < this.getInventory().getItemsCount(); i++)
 		{
-			ammoStocked += invItem.getQuantity();
+			CBlob@ invItem = this.getInventory().getItem(i);
```

Comment:
Avoid getting things like the inventory multiple times like this. Get it once before the loop and store it in a variable.

Reply:
It was like this before the PR too, but I've fixed it now